test/puppetlabs/puppetdb/testutils/events.clj

```diff
@@ -71,6 +73,25 @@
   [version example-resource-events report]
   (set (raw-expected-resource-events version example-resource-events report)))
 
+(defn query-resource-events
+  "Queries resource events and unstreams, used mainly for testing.
+
+  This wraps the existing streaming query code but returns results
+  and count (if supplied)."
+  [version query-sql]
+  {:pre [(map? query-sql)]}
+  (let [{[sql & params] :results-query
+         count-query    :count-query
+         projections    :projections} query-sql
+         result {:result (query/streamed-query-result
+                          version sql params
+                          ;; The doall simply forces the seq to be traversed
+                          ;; fully.
+                          (comp doall (events/munge-result-rows version nil)))}]
+    (if count-query
+      (assoc result :count (jdbc/get-result-count count-query))
+      result)))
+
 (defn resource-events-query-result
```

Comment:
@wkalt also maybe misunderstood, but thought we might be moving away from aligned let values.